### ISSUE TITLE
Sourcery suggests refactorings when viewing a diff

### ISSUE THREAD

**Issue by mjpieters** [sourcery-ai/sourcery-vscode#61]
Sourcery suggests refactorings for Python code shown in the [Visual Studio code diff view](https://code.visualstudio.com/docs/editor/versioncontrol#_viewing-diffs), usually for a refactoring _already applied_.

To reproduce, create a new Python project with an SCM (like git) or open an existing Python project from GitHub. Make a change in a file containing code that Sourcery could refactor, and go to the *Source Control* view (<kbd>⌘</kbd>+<kbd>P</kbd>, "Focus on Source Control View") and select the changed file to open the diff view. Then open the *Problems* panel (<kbd>⌘</kbd>+<kbd>P</kbd>, "Focus on Problems View") to see that Source Control has identified the refactoring opportunity.

Here is a specific test case:

1. Create /tmp/mcve
2. Open /tmp/mvce in Visual Studio Code in a new window.
3. Use the Source Control panel to initialise the directory as a git repository
4. Add a new file, `mcve.py` and add the 'nested if' example from the documentation, and save the file:

  ```python
  def merge_nested_if(a, b):
      if a:
          if b:
              return c
  ```
5. Stage this file (<kbd>⌘</kbd>+<kbd>P</kbd>, "Git: Stage changes")
6. Apply the proposed refactoring (<kbd>⌘</kbd>+<kbd>.</kbd>, "Merge nested if conditions")
7. Open the problems panel (<kbd>⌘</kbd>+<kbd>P</kbd>, "Focus on Problems View"). I use the [Pylance extension](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance), which identifies the `c` variable as undefined, no other issues are reported.
7. Go to the source control panel (<kbd>⌘</kbd>+<kbd>P</kbd>, "Focus on Source Control View")
8. Select `mcve.py` under *Changes*.
9. Observe that the (still open) Problems panel now shows sourcery is suggesting we merge the nested if conditions again.

Screenshot:

<img width="1071" alt="image" src="https://user-images.githubusercontent.com/46775/149510710-feac5880-2d1b-4cad-a0fb-76ccee53d488.png">

You can compare this to the staged change, which includes the Intellisense underline:

<img width="1071" alt="image" src="https://user-images.githubusercontent.com/46775/149511460-636a2393-1dd2-47e8-81cf-efacffde031f.png">

The Sourcery entry in the panel links to a read-only view of the file in the state it was staged at.

Sourcery should probably ignore read-only or diff views.

* Sourcery Version: 0.10.0
* Code editor or IDE name and version: Visual Studio Code 1.63.2
* OS name and version: OSX (macOS Big Sur 11.6.1)

**Comment by mjpieters** [comment]
I can fix this issue by updating the `documentSelector` definition in the extension:

https://github.com/sourcery-ai/sourcery-vscode/blob/main/src/extension.ts#L40

Replacing `'python'` with `{language: 'python', scheme: 'file'}` in my local copy of the extension resolves this issue. This would mean sourcery won't be able to make suggestions for new Python files that have not yet been saved to disk, but it would not affect Python files that are being edited (have changes that have not yet been saved). 

Also see the [Visual Studio Code documentation on document selectors](https://code.visualstudio.com/api/references/document-selector).

**Comment by Hellebore** [comment]
Thanks for raising, and pointing out the document selector docs!

**Comment by Hellebore** [comment]
This should now be addressed by 0.10.1
